### PR TITLE
Generate JVM version stats per month

### PR DESCRIPTION
**Pull request by batmat**
This change retrieves JVM informations, and for now only generates a
simple count of occurrences per month and JDK Major.Minor version.

Example:

```
{
    "jvmStatsPerMonth": {
        "1228086000000": {
            "1.5": 395,
            "1.6": 129,
            "1.7": 1
        },
        "1230764400000": {
            "1.5": 1544,
            "1.6": 1092,
            "1.7": 5
        },
...
```

Full example (with old dataset): https://gist.github.com/batmat/28361c7cadb0e134fbb8dd1c655190a1

Thanks for review @imod @orrc @daniel-beck @rtyler @oleg-nenashev @abayer

### CODE REVIEW

**Comment by rtyler** [comment]
Once the PR build completes I'll merge.

**Comment by batmat** [comment]
@rtyler it's unclear to me why the PR build failed. Seems like it failed before reaching my add-on, while generating the per-plugin stats.

**Comment by imod** [comment]
👍 
I'm not sure why the build failed, but I guess its related to:
 `Resuming build at Thu Oct 27 00:40:47 UTC 2016 after Jenkins restart`

**Comment by svanoort** [comment]
Love it!

**Comment by daniel-beck** [comment]
@batmat Forgot to adapt https://github.com/jenkins-infra/infra-statistics/blob/master/generate-index.sh#L51

Also, not clear why this is in `plugin-installation-trend`…

File: createJson.groovy [hunk 1]
```diff
@@ -175,6 +175,33 @@ class Generator {
         }
     }
 
+    // for each month, counts the number of JVM versions in use (using strict filtering to ignore weird/local JVM version names)
+    def generateJvmJson() {
+        final def JVM_VERSIONS = ["1.5", "1.6", "1.7", "1.8", "1.9"]
```
Comment (batmat) [inline createJson.groovy]: time to adjust this...

Comment (daniel-beck) [reply createJson.groovy]: @abayer hasn't yet run January's stats, so you could provide a patch.

Comment (batmat) [reply createJson.groovy]: https://github.com/jenkins-infra/infra-statistics/pull/42

Comment (batmat) [reply createJson.groovy]: OMG, answer is always 42 right?

**Pull request merged**